src/app/drive/services/file.service/uploadFile.ts

```diff
@@ -20,13 +23,64 @@ export interface FileUploadOptions {
   isUploadedFromFolder?: boolean;
 }
 
-class RetryableFileError extends Error {
-  constructor(public file: FileToUpload) {
-    super('Retryable file');
-    this.name = 'RetryableFileError';
-  }
+interface UploadFileProps {
+  file: FileToUpload;
+  fileId?: string;
+  isWorkspaceUpload?: boolean;
+  bucketId: string;
+  workspaceId?: string;
+  resourcesToken?: string;
+  ownerToken?: string;
 }
 
+export const createFileEntry = async ({
+  bucketId,
+  fileId,
+  file,
+  isWorkspaceUpload,
+  workspaceId,
+  resourcesToken,
+  ownerToken,
+}: UploadFileProps) => {
+  const date = new Date();
+
+  if (isWorkspaceUpload && workspaceId) {
+    if (!fileId) {
+      throw new FileIdRequiredError();
+    }
```

Comment:
I left this comment in order to check if the upload of empty files are already implemented in backend for workspaces in order to remove this check